linkerd/app/core/src/config.rs

```diff
@@ -1,1137 +1,115 @@
-            suffixes.push(sfx);
-        }
-    }
-
-    Ok(suffixes)
+#[derive(Clone, Debug)]
+pub struct DnsConfig {
+    pub min_ttl: Option<Duration>,
+    pub max_ttl: Option<Duration>,
+    pub resolv_conf_path: PathBuf,
 }
 
-fn parse_dns_suffix(s: &str) -> Result<dns::Suffix, ParseError> {
-    if s == "." {
-        return Ok(dns::Suffix::Root);
-    }
-
-    dns::Name::try_from(s.as_bytes())
-        .map(dns::Suffix::Name)
-        .map_err(|_| ParseError::NotADomainSuffix)
+#[derive(Clone, Debug)]
+pub struct IdentityConfig {
+    pub control: ControlConfig,
+    pub identity: identity::Config,
 }
 
-pub fn parse_backoff<S: Strings>(
-    strings: &S,
-    base: &str,
-    default: ExponentialBackoff,
-) -> Result<ExponentialBackoff, Error> {
-    let min_env = format!("LINKERD2_PROXY_{}_EXP_BACKOFF_MIN", base);
-    let min = parse(strings, &min_env, parse_duration);
-    let max_env = format!("LINKERD2_PROXY_{}_EXP_BACKOFF_MAX", base);
-    let max = parse(strings, &max_env, parse_duration);
-    let jitter_env = format!("LINKERD2_PROXY_{}_EXP_BACKOFF_JITTER", base);
-    let jitter = parse(strings, &jitter_env, parse_number::<f64>);
+// === impl ServerConfig ===
 
-    match (min?, max?, jitter?) {
-        (None, None, None) => Ok(default),
-        (Some(min), Some(max), jitter) => {
-            ExponentialBackoff::new(min, max, jitter.unwrap_or_default()).map_err(|error| {
-                error!(message="Invalid backoff", %error, %min_env, ?min, %max_env, ?max, %jitter_env, ?jitter);
-                Error::InvalidEnvVar
-            })
-        }
-        _ => {
-            error!("You need to specify either all of {} {} {} or none of them to use the default backoff", min_env, max_env,jitter_env );
-            Err(Error::InvalidEnvVar)
-        }
-    }
-}
-
-pub fn parse_control_addr<S: Strings>(
-    strings: &S,
-    base: &str,
-) -> Result<Option<ControlAddr>, Error> {
-    let a_env = format!("{}_ADDR", base);
-    let a = parse(strings, &a_env, parse_addr);
-    let n_env = format!("{}_NAME", base);
-    let n = parse(strings, &n_env, parse_identity);
-    match (a?, n?) {
-        (None, None) => Ok(None),
-        (Some(ref addr), _) if addr.is_loopback() => Ok(Some(ControlAddr {
-            addr: addr.clone(),
-            identity: Conditional::None(tls::ReasonForNoPeerName::Loopback.into()),
-        })),
-        (Some(addr), Some(name)) => Ok(Some(ControlAddr {
-            addr,
-            identity: Conditional::Some(name),
-        })),
-        (Some(_), None) => {
-            error!("{} must be specified when {} is set", n_env, a_env);
-            Err(Error::InvalidEnvVar)
-        }
-        (None, Some(_)) => {
-            error!("{} must be specified when {} is set", a_env, n_env);
-            Err(Error::InvalidEnvVar)
+impl<O: OrigDstAddr> ServerConfig<O> {
+    pub fn with_orig_dst_addrs_from<P: OrigDstAddr>(self, orig_dst_addrs: P) -> ServerConfig<P> {
```

Comment:
TIOLI: this name feels pretty wordy and `with_orig_dst` seems equally descriptive to me? Not important to me though.